### PR TITLE
Coordinator

### PR DESCRIPTION
**Pull request by gdey**
#17 #38 This gets the final parts for queueing.

### CODE REVIEW

File: atlante/server/coordinator/field/status.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		Error  error `json:"error"`
+	},
+}
+*/
+
+type (
+
+	// Status is used to hold a status for searilization
+	Status struct {
+		Status StatusEnum
```
Comment (ARolek) [inline atlante/server/coordinator/field/status.go]: Can we name our way around this stutter?

File: atlante/server/coordinator/field/status.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	if err := json.Unmarshal(b, &obj); err != nil {
+		return err
+	}
+	var typ string
+	if err := json.Unmarshal(obj["status"], &typ); err != nil {
+		return err
+	}
+
+	switch typ {
+	case "started":
```
Comment (ARolek) [inline atlante/server/coordinator/field/status.go]: use constants.

File: atlante/server/coordinator/field/status.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+		s.Status = Completed{}
+
+	default:
+		return fmt.Errorf("Unknown status type: %v", typ)
+
+	}
+	return nil
+}
+
+func NewStatusFor(s, desc string) (StatusEnum, error) {
```
Comment (ARolek) [inline atlante/server/coordinator/field/status.go]: don't use single letter vars. I think s should be status?

File: atlante/server/coordinator/field/status.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+	switch strings.ToLower(s) {
+	case "started":
```
Comment (ARolek) [inline atlante/server/coordinator/field/status.go]: use consts

File: atlante/server/coordinator/logger/logger.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+			return errors.String("unknown field type")
+		}
+	}
+	if p == nil || p.Provider != nil {
+		return p.Provider.UpdateField(job, fields...)
+	}
+	return nil
+}
+
+func (p *Provider) FindJob(job *atlante.Job) (jb *coordinator.Job, found bool) {
```
Comment (ARolek) [inline atlante/server/coordinator/logger/logger.go]: Potentially change name and add some comments.

File: atlante/server/coordinator/postgresql/README.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+* `ssl_key` (string) : [optional] the ssl key for postgres SSL
+* `ssl_cert` (string) : [optional] the ssl cert for postgres SSL
+* `ssl_root_cert` (string) : [optional] the ssl root cert
+* `max_connections` (number) : [optional] the max number of connections to keep in the pool
+
+## SQL options.
+The following options are used to specify the sql that is used to manage the 
+database. (note this can be brittle.)
+
+* `query_new_job` : the sql is run to create a new job.
```
Comment (ARolek) [inline atlante/server/coordinator/postgresql/README.md]: document param type.

File: atlante/server/coordinator/postgresql/README.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+* `port` (string) : [required] the database post
+* `user` (string) : [required] the user for the database
+* `password` (string) : [required] the user password
+* `ssl_mode` (string) : [optional] the ssl mode for postgres SSL
+* `ssl_key` (string) : [optional] the ssl key for postgres SSL
+* `ssl_cert` (string) : [optional] the ssl cert for postgres SSL
+* `ssl_root_cert` (string) : [optional] the ssl root cert
+* `max_connections` (number) : [optional] the max number of connections to keep in the pool
+
+## SQL options.
```
Comment (ARolek) [inline atlante/server/coordinator/postgresql/README.md]: Document the table definitions that are used by the default queries.

File: atlante/server/coordinator/postgresql/docs/jobs.sql [hunk 8]
```diff
@@ -0,0 +1,49 @@
+CREATE EXTENSION postgis;
```
Comment (ARolek) [inline atlante/server/coordinator/postgresql/docs/jobs.sql]: can we make this not fail if postgis is already instantiated?

File: atlante/server/coordinator/postgresql/docs/jobs.sql [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+CREATE TABLE jobs (
+    id SERIAL PRIMARY KEY,
+    mdgid TEXT NOT NULL,
+    sheet_number integer DEFAULT 0,
+    sheet_name TEXT NOT NULL,
+    queue_id TEXT,
+    job_data TEXT,
+    bounds GEOMETRY,
```
Comment (ARolek) [inline atlante/server/coordinator/postgresql/docs/jobs.sql]: Use polygon type (i.e. `bounds geometry(Polygon,4326) NOT NULL`)

File: atlante/server/coordinator/field/status.go (hunk 1 above)
Comment (gdey) [reply atlante/server/coordinator/field/status.go]: Leaving in Stutter till we have a better idea. Created issue #45  for this.

**Pull request merged**